src/main/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java

```diff
@@ -103,39 +102,63 @@ public static void main(String... args)
      * @param inputFile .java file
      * @param outputFile .properties file
      * @throws CheckstyleException if a javadoc comment can not be parsed
-     * @throws IOException if there is a problem with files access
+     * @noinspection NestedTryStatement, OverlyNestedMethod
```

Comment:
no.
it is very rare case, when we are ok to make suppressions.

I played a bit with your code and here is my proposal, to split streams mess and pure business logic of generator:
```
    private static void writePropertiesFile(String inputFile, String outputFile)
            throws CheckstyleException {
        FileOutputStream fos = null;
        PrintWriter writer = null;
        try {
            fos = new FileOutputStream(outputFile);
            final Writer osw = new OutputStreamWriter(fos, StandardCharsets.UTF_8);
            writer = new PrintWriter(osw, false);

            final DetailAST top = JavaParser.parseFile(new File(inputFile),
                JavaParser.Options.WITH_COMMENTS);
            iteratePublicStaticIntFields(top, writer::println);

        }
        catch (IOException ex) {
            throw new CheckstyleException("Failed to write javadoc properties of '" + inputFile
                + "' to '" + outputFile + "'", ex);
        }
        finally {
            CommonUtils.close(writer);
            CommonUtils.close(fos);
        }
    }

    private static void iteratePublicStaticIntFields(DetailAST top, Consumer<String> writer)
            throws CheckstyleException {
        final DetailAST objBlock = getClassBody(top);
        if (objBlock != null) {
            for (DetailAST member = objBlock.getFirstChild(); member != null;
                 member = member.getNextSibling()) {
                if (isPublicStaticFinalIntField(member)) {
                    final DetailAST modifiers = member.findFirstToken(TokenTypes.MODIFIERS);
                    final String firstJavadocSentence = getFirstJavadocSentence(modifiers);
                    if (firstJavadocSentence != null) {
                        writer.accept(getName(member) + "=" + firstJavadocSentence.trim());
                    }
                }
            }
        }
    }
```

it will trigger inspection violation, but it is ok to update inspection config to allow:
![image](https://user-images.githubusercontent.com/812984/36156485-085224b4-108c-11e8-94c3-1efe41066ac7.png)
`<option name="insideTryAllowed" value="true" />` at config/intellij-idea-inspections.xml file.
here is example of violation and misunderstanding from community - https://intellij-support.jetbrains.com/hc/en-us/community/posts/206242079--OutputStreamWriter-should-be-opened-in-front-of-a-try-block-and-closed-in-the-corresponding-finally-block

Reply:
I wish I could write `objBlock.stream().filter(Class::isPublicStaticFinalIntField).map(Class::getFirstSentence).filter(Objects::nonNull).forEach(line -> writer.println(ln));`

done, by the way.